packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.types.ts

```diff
@@ -122,6 +122,12 @@ export interface ICheckboxProps extends React.ButtonHTMLAttributes<HTMLElement |
    * Optional keytip for this checkbox
    */
   keytipProps?: IKeytipProps;
+
+  /**
+   * Optional indeterminate state for this checkbox.
+   * Setting indeterminate state takes precedence over checked and defaultChecked props given.
+   */
+  indeterminate?: boolean;
```

Comment:
Do we allow to control this prop? Meaning that can we toggle on from the consuming app the indeterminate state after the first click? I would assume not. If so, worth mentioning that in the docs.

Reply:
Correct, It is applied to the checkbox on load but then disappears after the first click without reappearing under any circumstance. Thanks, I'll add that to the docs!

Reply:
This seems consistent with other apis. All of them use `indeterminate` as an overridden value. Semantic uses `defaultIndeterminate` to support uncontrolled modes, but it is the only one here afaik.

https://material-ui.com/api/checkbox
https://react.semantic-ui.com/modules/checkbox/#states-indeterminate
https://ant.design/components/checkbox/
https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/Checkbox/Checkbox.js

Reply:
Wait. I might have read your comment wrong @maxwellred . I think if you have `indeterminate` flag on, and you click on the checkbox, it stays in an indeterminate state until the caller reacts to `onChange` and removes it.

If you want an uncontrolled behavior, I suggest `defaultIndeterminate` to be symmetrical with `defaultChecked`.

Reply:
Hi @dzearing i've set up indeterminate state to work as you describe where the dev passes in the indeterminate prop which sets isChecked to undefined and keeps isIndeterminate state set to true until onChange (per successive click) removes it. 

I'm a little confused as to how an uncontrolled behavior is to be implemented as the developer has to manually place the indeterminate props on components. Is the intention that we place a native indeterminate attribute on the Checkbox's <input> tag?

Reply:
@maxwellred I agree with @dzearing here. Atm your `indeterminate` prop, per its JSDoc, is supposed to behave like an uncontrolled prop (i.e. what `defaultIndeterminate` would have done) except that at the same time, it behaves like a controlled prop (because you update state in `getDerivedStateFromProps`).

If you could add in an example in the documentation state for using indeterminate checkbox in a controlled fashion, you should be able to see what the problem is (which is that as you control the checked/unchecked state of the component, the checkbox keeps flipping (or staying) in indeterminate .. violating what is implied in the JSDoc about how this property works.

It is probably cleaner and less confusing if we instead have dedicated `indeterminate` and `defaultIndeterminate` (mirroring checked and defaultChecked)  prop - if you want to use checkbox in a controlled fashion, use "checked" and "indeterminate" together. Otherwise, use defaultIndeterminate + defaultChecked.

Reply:
@cliffkoh @dzearing Alright I updated the functionality to mirror checked/defaultChecked, documentation, and the aria-checked attribute. I changed the functionality of the indeterminate state to be only visual and not affect checked state. My assumption was that since indeterminate state is an on load operation that the checkbox ought to be in this undefined state however the spec goes into how upon removing indeterminate state the checkbox should return the previous state it was set on (my bad).